File: Typscript/Step-1 Basic/Maths/armstrong.ts
// Time Complexity: O(2Log10(N))

function isArmstrong(n:number):boolean {
    let dup = n;
    let armstrong = 0;
    const noOfDigits = Math.floor(Math.log10(n))+1
    while(n > 0) {
        let lastDigit = n%10;
        armstrong += Math.pow(lastDigit,noOfDigits);
        n = Math.floor(n/10);
    }
    return dup == armstrong
}

console.log(isArmstrong(1634))

File: Go/Step1/Maths/armstrong.go
package main

import "math"

func isArmstrong(num int) bool {
	noOfDigits := int(math.Log10((float64(num)))) + 1
	dup := num
	armstrong := 0
	for num > 0 {
		lastDigit := num % 10
		armstrong = armstrong + int(math.Pow(float64(lastDigit), float64(noOfDigits)))
		num = num / 10
	}

	if armstrong == dup {
		return true
	}
	return false
}
